### ISSUE TITLE
Activate `clippy::manual_let_else` lint

### ISSUE THREAD

**Issue by eserilev** [sigp/lighthouse#4888]
## Description

There are several instances of manual `let else` usage within the Lighthouse codebase. Clippy has a lint that warns on manual `let else` usage when the `match` has only one other arm: https://rust-lang.github.io/rust-clippy/master/index.html#/manual_let_else

It would be nice to add this lint and refactor accordingly as it can improve code readability & aligns w/ idiomatic usage of Rust

## Example

https://github.com/sigp/lighthouse/blob/441fc1691b69f9edc4bbdc6665f3efab16265c9b/beacon_node/http_api/src/sync_committees.rs#L33-L38

can be refactored to

```rust
let Some(altair_fork_epoch) = chain.spec.altair_fork_epoch else {
  // Empty response for networks with Altair disabled.
  return Ok(convert_to_response(vec![], false));
};
```

## Steps to resolve

Enable `clippy::manual_let_else` lint and resolve the warning messages.

**Comment by eserilev** [comment]
I can pick this task up if this is a lint rule the Lighthouse team wants to add

**Comment by michaelsproul** [comment]
Yeah I think this would be cool. You might even be able to partly automate it with `make lint-fix` (after adding `-D clippy::manual_let_else` I guess)

**Comment by chong-he** [comment]
Solved in #4889
